packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -771,6 +844,36 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
     return _currentBottomSheet;
   }
 
+  // FAB API
+  AnimationController _fabMoveController;
+  FabPosition _previousFabPosition;
+  FabPosition _fabPosition;
+
+  void _moveFab(final FabPosition newPosition) {
+    void updatePosition() {
+      setState(() {
+        _previousFabPosition = _fabPosition;
```

Comment:
what if the previous animation didn't complete?
we should probably set _previousFabPosition to its actual current position.

Reply:
I've reworked this section to schedule animations more intelligently.